prov/efa/src/efa_av.c

```diff
@@ -124,6 +125,18 @@ void efa_rdm_peer_clear(struct rxr_ep *ep, struct rdm_peer *peer)
 	if (peer->robuf.pending)
 		ofi_recvwin_free(&peer->robuf);
 
+	/* we cannot release outstanding TX packets because device
+	 * will report completion of these packets later. Setting
+	 * the address to FI_ADDR_NOTAVAIL, so rxr_ep_get_peer()
+	 * will return NULL for the address, so the completion will
+	 * be ignored.
+	 */
+	dlist_foreach_container(&peer->outstanding_tx_pkts,
+				struct rxr_pkt_entry,
+				pkt_entry, entry) {
+		pkt_entry->addr = FI_ADDR_NOTAVAIL;
```

Comment:
At both the `rxr_pkt_entry` structure and the places where we check for `FI_ADDR_NOTAVAIL` as the address, please add comments about this case.  Why isn't there any checking for this case?

Reply:
All packet handler call function `rxr_ep_get_peer` to convert `pkt_entry->addr` to pointer to `peer`.

The function `rxr_ep_get_peer`, which will return  NULL if address is `FI_ADD_NOTAVAIL`.

Packet handler then check whether peer is NULL.

To avoid confusion, it seems to be better that packet handler check `pkt_entry->addr` against `FI_ADDR_NOTAVAIL` before calling `rxr_ep_get_peer`.

Reply:
i'm not complaining about what the code does, I'm just asking that you document all the impacted places.

Reply:
Updated comments to the structure rx_pkt_entry.